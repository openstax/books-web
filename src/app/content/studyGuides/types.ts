import { HighlightsSummary } from '@openstax/highlighter/dist/api';

export interface State {
  isEnabled: boolean;
  summary: HighlightsSummary | null;
}
