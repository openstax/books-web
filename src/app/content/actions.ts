import { createStandardAction } from 'typesafe-actions';
import * as highlightingActions from './highlights/actions';
import * as searchActions from './search/actions';
import { ArchivePage, Book, Params, State } from './types';

export const search = searchActions;
export const highlighting = highlightingActions;

export const openToc = createStandardAction('Content/openToc')<void>();
export const closeToc = createStandardAction('Content/closeToc')<void>();
export const resetToc = createStandardAction('Content/resetToc')<void>();

export const requestBook = createStandardAction('Content/requestBook')<Params['book']>();
export const receiveBook = createStandardAction('Content/receiveBook')<Book>();

export const requestPage = createStandardAction('Content/requestPage')<Params['page']>();
export const receivePage = createStandardAction('Content/receivePage')<
  ArchivePage & {references: State['references']}
>();

export const closeCallToActionPopup = createStandardAction('Content/closeCallToActionPopup')();
export const openCallToActionPopup = createStandardAction('Content/openCallToActionPopup')();
