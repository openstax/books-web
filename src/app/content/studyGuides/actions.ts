import { createStandardAction } from 'typesafe-actions';
import {
  CountsPerSource,
  SummaryFilters,
  SummaryHighlights,
  SummaryHighlightsPagination
} from '../highlights/types';

export const receiveSummaryStudyGuides = createStandardAction(
  'Content/StudyGuides/Summary/receive'
)<
SummaryHighlights,
  {
    pagination: SummaryHighlightsPagination,
    isStillLoading?: boolean
  }
>();

export const openStudyGuides = createStandardAction('Content/StudyGuides/Summary/open')<void>();
export const closeStudyGuides = createStandardAction('Content/StudyGuides/Summary/close')<void>();
export const loadMoreStudyGuides = createStandardAction('Content/StudyGuides/loadMore')();
export const setDefaultSummaryFilters = createStandardAction('Content/StudyGuides/Summary/setDefaultFilters')<
  Partial<SummaryFilters>
>();
export const setSummaryFilters = createStandardAction('Content/StudyGuides/Summary/setFilters')<
  Partial<SummaryFilters>
>();
export const receiveStudyGuidesTotalCounts = createStandardAction(
  'Content/StudyGuides/receiveTotalCounts'
)<CountsPerSource>();
