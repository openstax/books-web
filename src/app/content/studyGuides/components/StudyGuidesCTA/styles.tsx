import styled, { css } from 'styled-components/macro';
import Button from '../../../../components/Button';
import theme from '../../../../theme';

// tslint:disable-next-line: variable-name
export const StudyGuidesCTAWrapper = styled.div`
  display: flex;
  height: 20rem;
  padding: 0 8rem;
  justify-content: space-between;
  align-items: center;
  color: ${theme.color.text.default};
  box-shadow: inset 0 0 10px 0 rgba(0, 0, 0, 0.4);
  overflow: hidden;

  strong {
    color: ${theme.color.secondary.deepGreen.base};
  }

  ${theme.breakpoints.mobile(css`
    height: 17.6rem;
    padding: 0 1.6rem;
  `)}
`;

// tslint:disable-next-line: variable-name
export const StudyGuidesCTAContent = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
`;

// tslint:disable-next-line: variable-name
export const StudyGuidesCTATitle = styled.h2`
  font-size: 2.4rem;
  margin: 0 0 1rem 0;
  ${theme.breakpoints.mobile(css`
    font-size: 1.8rem;
    line-height: 2.6rem;
  `)}
`;

// tslint:disable-next-line: variable-name
export const StudyGuidesCTAButtons = styled.div`
  display: flex;
  align-items: center;
  flex-wrap: wrap;
`;

// tslint:disable-next-line: variable-name
export const StudyGuidesCTAButton = styled(Button)`
  width: 24rem;
  height: 4.8rem;
  padding: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 1.8rem;
  font-weight: 600;
  color: ${theme.color.text.white};
  background-color: ${theme.color.secondary.deepGreen.base};
  border: none;
  margin-right: 1.6rem;

  &:hover {
    background-color: ${theme.color.secondary.deepGreen.base};
  }

  ${theme.breakpoints.mobile(css`
    font-size: 1.4rem;
    width: 11rem;
    height: 3rem;
  `)}
`;

// tslint:disable-next-line: variable-name
export const StudyGuidesCTAButtonsSecondary = styled.div`
  padding: 1.4rem 0;
`;

// tslint:disable-next-line: variable-name
export const StudyGuidesCTASeparator = styled.span`
  font-size: 1.6rem;
  text-transform: uppercase;
  margin-right: 1.6rem;
  ${theme.breakpoints.mobile(css`
    font-size: 1.4rem;
    margin-right: 1.2rem;
  `)}
`;

// tslint:disable-next-line: variable-name
export const StudyGuidesCTALink = styled.a`
  font-size: 1.6rem;
  color: #027eb5;
  ${theme.breakpoints.mobile(css`
    font-size: 1.4rem;
  `)}
`;

// tslint:disable-next-line: variable-name
export const StudyGuidesCTAInfoWrapper = styled.div`
  flex: 1;
  display: flex;
  justify-content: center;
  height: 100%;
  text-align: center;
  overflow: visible;
  position: relative;
  ${theme.breakpoints.mobile(css`
    justify-content: flex-end;
  `)}
`;

// tslint:disable-next-line: variable-name
export const StudyGuidesCTAInfo = styled.div`
  font-size: 1.8rem;
  line-height: 2.5rem;
  width: 24rem;
  margin-top: 2.5rem;
  z-index: 1;

  strong {
    display: block;
  }

  ${theme.breakpoints.mobile(css`
    width: 10rem;
    font-size: 1.2rem;
    line-height: 1.7rem;
    margin-top: 1.6rem;
    margin-right: 3.4rem;
  `)}
`;

// tslint:disable-next-line: variable-name
export const StudyGuidesCTAArrowDesktop = styled.img`
  position: absolute;
  ${theme.breakpoints.mobile(css`
    display: none;
  `)}
`;

// tslint:disable-next-line: variable-name
export const StudyGuidesCTAArrowMobile = styled.img`
  position: absolute;
  right: -2.5rem;
  display: none;
  ${theme.breakpoints.mobile(css`
    display: block;
  `)}
`;
