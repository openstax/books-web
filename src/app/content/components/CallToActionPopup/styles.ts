import styled from 'styled-components';
import { Times } from 'styled-icons/fa-solid/Times/Times';
import Button from '../../../components/Button';
import { textRegularStyle } from '../../../components/Typography';
import { textStyle } from '../../../components/Typography/base';
import theme from '../../../theme';

// tslint:disable-next-line: variable-name
export const CTAWrapper = styled.div`
  position: fixed;
  left: 0;
  bottom: 0;
  z-index: 99;
  width: 100%;

  @media screen and (max-width: 1200px) {
    display: none;
  }
`;

// tslint:disable-next-line: variable-name
export const CTAContent = styled.div`
  position: absolute;
  z-index: 3;
  bottom: 0;
  width: 100%;
  display flex;
  justify-content: center;
  align-items: center;
`;

// tslint:disable-next-line: variable-name
export const CTAText = styled.div``;

// tslint:disable-next-line: variable-name
export const CTAHeading = styled.h2`
  ${textStyle}
  font-size: 3.6rem;
  line-height: 4rem;
  color: ${theme.color.text.white};
  margin: 0;
  max-width: 570px;
  overflow: hidden;
`;

// tslint:disable-next-line: variable-name
export const CTAParagraph = styled.p`
  ${textStyle}
  font-size: 1.8rem;
  line-height: 3rem;
  margin: 1rem 0;
  color: ${theme.color.text.white};
  max-width: 570px;
`;

// tslint:disable-next-line: variable-name
export const CTAButtons = styled.div`
  display: flex;
  align-items: center;
  ${Button} {
    width: 170px;
  }
`;

// tslint:disable-next-line: variable-name
export const CTATextLink = styled.span`
  ${textRegularStyle}
  color: ${theme.color.text.white};
  margin-left: 1.5rem;
  a {
    color: ${theme.color.text.white};
    text-decoration: underline;
    margin-left: 0.5rem;
  }
`;

// tslint:disable-next-line: variable-name
export const CTACloseButton = styled(Times)`
  position: absolute;
  top: 40%;
  right: 3rem;
  z-index: 3;
  width: 1.8rem;
  color: ${theme.color.text.white};
  cursor: pointer;
`;

// tslint:disable-next-line: variable-name
export const CTAGraphic = styled.img`
  margin-left: 3rem;
`;

// tslint:disable-next-line: variable-name
export const CTATopLayer = styled.img`
  position: absolute;
  right: -5px;
  bottom: 0;
  z-index: 2;
`;

// tslint:disable-next-line: variable-name
export const CTAMiddleLayer = styled.img`
  width: 100%;
  position: absolute;
  left: 0;
  bottom: 0;
  z-index: 1;
`;

// tslint:disable-next-line: variable-name
export const CTABottomLayer = styled.img`
  width: 100%;
  position: absolute;
  left: 0;
  bottom: 20px;
  z-index: 0;
`;

// tslint:disable-next-line: variable-name
export const CTABackground = styled.div`
  position: relative;
  height: 420px;
  overflow: hidden;
`;
