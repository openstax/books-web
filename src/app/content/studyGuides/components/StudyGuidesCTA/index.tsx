import React from 'react';
import { FormattedMessage } from 'react-intl';
import { useSelector } from 'react-redux';
import {
  loggedOut,
  loginLink as loginLinkSelector,
  signupLink as signupLinkSelector,
} from '../../../../auth/selectors';
import htmlMessage from '../../../../components/htmlMessage';
import arrowDesktop from './assets/arrow-desktop.svg';
import arrowMobile from './assets/arrow-mobile.svg';
import * as Styled from './styles';

// tslint:disable-next-line:variable-name
const StudyGuidesCTATitle = htmlMessage('i18n:studyguides:cta:title', Styled.StudyGuidesCTATitle);

// tslint:disable-next-line:variable-name
const StudyGuidesCTAInfo = htmlMessage('i18n:studyguides:cta:info', Styled.StudyGuidesCTAInfo);

// tslint:disable-next-line: variable-name
const StudyGuidesCTA = () => {
  const isNotLoggedIn = useSelector(loggedOut);
  const loginLink = useSelector(loginLinkSelector);
  const signupLink = useSelector(signupLinkSelector);

  if (!isNotLoggedIn) { return null; }

  return <Styled.StudyGuidesCTAWrapper>
    <Styled.StudyGuidesCTAContent>
      <StudyGuidesCTATitle />
      <Styled.StudyGuidesCTAButtons>
        <FormattedMessage id='i18n:studyguides:cta:button'>
          {(msg: Element | string) => <Styled.StudyGuidesCTAButton
            // eslint-disable-next-line
            component={<a href={signupLink}/>}
          >{msg}</Styled.StudyGuidesCTAButton>}
        </FormattedMessage>
        <Styled.StudyGuidesCTAButtonsSecondary>
          <Styled.StudyGuidesCTASeparator>
            <FormattedMessage id='i18n:studyguides:cta:separator'>
              {(msg: Element | string) => msg}
            </FormattedMessage>
          </Styled.StudyGuidesCTASeparator>
          <FormattedMessage id='i18n:studyguides:cta:login'>
            {(msg: Element | string) => <Styled.StudyGuidesCTALink
              href={loginLink}
            >{msg}</Styled.StudyGuidesCTALink>}
          </FormattedMessage>
        </Styled.StudyGuidesCTAButtonsSecondary>
      </Styled.StudyGuidesCTAButtons>
    </Styled.StudyGuidesCTAContent>
    <Styled.StudyGuidesCTAInfoWrapper>
      <StudyGuidesCTAInfo />
      <Styled.StudyGuidesCTAArrowDesktop src={arrowDesktop} alt='' />
      <Styled.StudyGuidesCTAArrowMobile src={arrowMobile} alt='' />
    </Styled.StudyGuidesCTAInfoWrapper>
  </Styled.StudyGuidesCTAWrapper>;
};

export default StudyGuidesCTA;
