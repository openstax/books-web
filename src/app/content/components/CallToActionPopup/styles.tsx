import React from 'react';
import styled, { css, keyframes } from 'styled-components';
import { Times } from 'styled-icons/fa-solid/Times/Times';
import Button from '../../../components/Button';
import { textRegularStyle } from '../../../components/Typography';
import { textStyle } from '../../../components/Typography/base';
import theme from '../../../theme';
import bottomLayer from './assets/bottom-layer.svg';
import ctaGraphic from './assets/desktop-mobile-graphic.svg';
import middleLayer from './assets/middle-layer.svg';
import topLayer from './assets/top-layer.svg';

const slideIn = keyframes`
  0% {
    bottom: -100%;
  }

  100% {
    bottom: 0;
  }
`;

const slideInAnimation = css`
  animation: ${300}ms ${slideIn} ease-out;
`;

// tslint:disable-next-line: variable-name
export const CTAWrapper = styled.div`
  position: fixed;
  left: 0;
  bottom: 0;
  z-index: ${theme.zIndex.ctaPopup};
  width: 100%;
  ${slideInAnimation}
  ${theme.breakpoints.mobile(css`
    display: none;
  `)}
`;

// tslint:disable-next-line: variable-name
export const CTAContent = styled.div`
  position: absolute;
  bottom: 0;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 1;
`;

// tslint:disable-next-line: variable-name
export const CTAHeading = styled.h2`
  ${textStyle}
  font-size: 3.6rem;
  line-height: 4rem;
  letter-spacing: -0.14rem;
  color: ${theme.color.text.white};
  margin: 0;
  max-width: 50rem;
  overflow: hidden;
`;

// tslint:disable-next-line: variable-name
export const CTAParagraph = styled.p`
  ${textStyle}
  font-size: 1.8rem;
  line-height: 3rem;
  margin: 1rem 0;
  color: ${theme.color.text.white};
  max-width: 57rem;
`;

// tslint:disable-next-line: variable-name
export const CTAButtons = styled.div`
  display: flex;
  align-items: center;

  ${Button} {
    width: 17rem;
  }
`;

// tslint:disable-next-line: variable-name
export const CTATextLink = styled.span`
  ${textRegularStyle}
  color: ${theme.color.text.white};
  margin-left: 1.5rem;

  a {
    color: ${theme.color.text.white};
    text-decoration: underline;
    margin-left: 0.5rem;
  }
`;

// tslint:disable-next-line: variable-name
export const CTACloseButton = styled(Button)`
  position: absolute;
  top: 47%;
  right: 1%;
  background: transparent;
  min-width: auto;
  padding: 2rem;
  border: none;
  z-index: 1;

  &:hover {
    background: transparent;
  }
`;

// tslint:disable-next-line: variable-name
export const CTACloseIcon = styled(Times)`
  width: 1.8rem;
  color: ${theme.color.text.white};
`;

// tslint:disable-next-line: variable-name
export const CTAGraphic = styled((props) => <img src={ctaGraphic} alt='' {...props} />)`
  margin-left: 3rem;
`;

// tslint:disable-next-line: variable-name
export const CTATopLayer = styled((props) => <img src={topLayer} alt='' {...props} />)`
  position: absolute;
  right: -0.5rem;
  bottom: 0;
`;

// tslint:disable-next-line: variable-name
export const CTAMiddleLayer = styled((props) => <img src={middleLayer} alt='' {...props} />)`
  width: 100%;
  min-width: 160rem;
  position: absolute;
  left: 0;
  bottom: 0;
`;

// tslint:disable-next-line: variable-name
export const CTABottomLayer = styled((props) => <img src={bottomLayer} alt='' {...props} />)`
  width: 100%;
  min-width: 160rem;
  position: absolute;
  left: 0;
  bottom: 2rem;
`;

// tslint:disable-next-line: variable-name
export const CTABackground = styled.div`
  position: relative;
  height: 42rem;
  overflow: hidden;
`;
