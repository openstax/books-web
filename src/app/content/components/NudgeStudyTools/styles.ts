import styled, { css } from 'styled-components/macro';
import { Times } from 'styled-icons/fa-solid/Times/Times';
import { PlainButton } from '../../../components/Button';
import htmlMessage from '../../../components/htmlMessage';
import theme from '../../../theme';
import { remsToPx } from '../../../utils';
import { arrowDesktopHeight, arrowMobileHeight, closeButtonMobileMargin } from './constants';

// tslint:disable-next-line: variable-name
export const NudgeWrapper = styled.div`
  display: contents;
`;

// tslint:disable-next-line: variable-name
export const NudgeContentWrapper = styled.div`
  position: fixed;
  z-index: ${theme.zIndex.nudgeOverlay + 1};
  outline: none;
  ${(props: { top: number, right: number }) => `
    top: ${props.top}px;
    right: ${props.right}px;
  `}
  ${theme.breakpoints.mobile(css`
    right: auto;
    width: 100%;
    text-align: center;
    padding: 0 2rem;
  `)}
`;

// tslint:disable-next-line: variable-name
export const NudgeContent = styled.div`
  position: relative;
`;

// tslint:disable-next-line: variable-name
const NudgeHeadingStyles = styled.h2`
  font-size: 3.6rem;
  line-height: 1.1;
  letter-spacing: -1.4px;
  margin: 0 0 1.7rem 0;
  color: ${theme.color.text.white};
  overflow: hidden;

  strong {
    color: ${theme.color.primary.yellow.base};
    font-weight: 600;
    white-space: nowrap;
  }

  ${theme.breakpoints.mobile(css`
    font-size: 2.4rem;
  `)}
`;

// tslint:disable-next-line: variable-name
export const NudgeHeading = htmlMessage('i18n:nudge:study-tools:heading', NudgeHeadingStyles);

// tslint:disable-next-line: variable-name
export const NudgeTextStyles = styled.div`
  font-size: 2.4rem;
  font-weight: 400;
  line-height: 1.1;
  letter-spacing: -1px;
  color: ${theme.color.text.white};
  max-width: 690px;
  overflow: hidden;
  ${theme.breakpoints.mobile(css`
    max-width: 100%;
    font-size: 1.6rem;
    line-height: 1.4;
    padding: 0 2rem;
  `)}
`;

// tslint:disable-next-line: variable-name
export const NudgeArrow = styled.img`
  position: fixed;
  z-index: ${theme.zIndex.nudgeOverlay + 1};
  display: block;
  height: ${arrowDesktopHeight}rem;
  ${(props: { top: number, left: number }) => `
    top: ${props.top}px;
    left: ${props.left}px;
  `}
  ${theme.breakpoints.mobile(css`
    height: ${arrowMobileHeight}rem;
  `)}
`;

// tslint:disable-next-line: variable-name
export const NudgeCloseIcon = styled(Times)`
  width: 1.1rem;
  color: ${theme.color.text.white};
`;

// tslint:disable-next-line: variable-name
export const NudgeCloseButton = styled(PlainButton)`
  position: fixed;
  z-index: ${theme.zIndex.nudgeOverlay + 1};
  ${(props: { top: number, left: number }) => `
    top: ${props.top}px;
    left: ${props.left}px;
  `}
  width: 4rem;
  height: 4rem;
  padding: 1rem;
  border-radius: 50%;
  border: 1px solid ${theme.color.white};
  display: flex;
  align-items: center;
  justify-content: center;

  ${theme.breakpoints.mobile(css`
    left: auto;
    top: ${remsToPx(closeButtonMobileMargin)}px;
    right: ${remsToPx(closeButtonMobileMargin)}px;
  `)}
`;

interface NudgeSpotlightPlacement {
  top: number;
  left: number;
  height: number;
  width: number;
}

// tslint:disable-next-line: variable-name
export const NudgeBackground = styled.div`
  position: fixed;
  top: 0;
  width: 100%;
  height: 100%;
  z-index: ${theme.zIndex.nudgeOverlay};
  pointer-events: none;

  display: grid;
  ${(props: NudgeSpotlightPlacement) => `
    grid-template: ${props.top}px ${props.height}px 1fr / ${props.left}px ${props.width}px 1fr;
    grid-template-areas:
      'top       top       top'
      'left   spotlight   right'
      'bottom   bottom   bottom';
  `}
`;

interface ClickBlockerProps {
  area: 'top' | 'right' | 'bottom' | 'left';
}

// tslint:disable-next-line: variable-name
export const ClickBlocker = styled.div`
  width: 100%;
  height: 100%;
  pointer-events: all;
  background-color: ${theme.color.black};
  opacity: 0.9;

  grid-area: ${({area}: ClickBlockerProps) => area};
`;
