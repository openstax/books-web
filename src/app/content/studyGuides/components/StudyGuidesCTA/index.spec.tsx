import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import StudyGuidesCTA from '.';
import createTestStore from '../../../../../test/createTestStore';
import { receiveLoggedOut, receiveUser } from '../../../../auth/actions';
import MessageProvider from '../../../../MessageProvider';
import { Store } from '../../../../types';
import * as Styled from './styles';

describe('StudyGuidesCTA', () => {
  let store: Store;

  beforeEach(() => {
    store = createTestStore();
  });

  it('does not render if user is logged in', () => {
    store.dispatch(receiveUser({} as any));

    const component = renderer.create(<Provider store={store}>
      <MessageProvider>
        <StudyGuidesCTA />
      </MessageProvider>
    </Provider>);

    expect(() => component.root.findByType(Styled.StudyGuidesCTAWrapper)).toThrow();
  });

  it('does render if user is not logged in', () => {
    store.dispatch(receiveLoggedOut());

    const component = renderer.create(<Provider store={store}>
      <MessageProvider>
        <StudyGuidesCTA />
      </MessageProvider>
    </Provider>);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
