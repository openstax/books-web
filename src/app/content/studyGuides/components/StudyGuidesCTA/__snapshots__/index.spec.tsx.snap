// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudyGuidesCTA does render if user is not logged in 1`] = `
<div
  className="styles__StudyGuidesCTAWrapper-dthmzf-0 kyjsks"
>
  <div
    className="styles__StudyGuidesCTAContent-dthmzf-1 jUYjMg"
  >
    <h2
      className="styles__StudyGuidesCTATitle-dthmzf-2 dSrkmG"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Expert-created study guides. <strong>100% free!</strong>",
        }
      }
    />
    <div
      className="styles__StudyGuidesCTAButtons-dthmzf-3 hwbGXN"
    >
      <a
        className="Button-ayg7nk-0 styles__StudyGuidesCTAButton-dthmzf-4 pleeo"
        href="/accounts/signup?r=/"
      >
        Sign Up
      </a>
      <div
        className="styles__StudyGuidesCTAButtonsSecondary-dthmzf-5 bJMpUy"
      >
        <span
          className="styles__StudyGuidesCTASeparator-dthmzf-6 kyyUHw"
        >
          or
        </span>
        <a
          className="styles__StudyGuidesCTALink-dthmzf-7 kAeWfN"
          href="/accounts/login?r=/"
        >
          Log in
        </a>
      </div>
    </div>
  </div>
  <div
    className="styles__StudyGuidesCTAInfoWrapper-dthmzf-8 ilsAaO"
  >
    <div
      className="styles__StudyGuidesCTAInfo-dthmzf-9 bjQynl"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Like this guide? <br>We've got study guides for every chapter. <strong>Free forever.</strong>",
        }
      }
    />
    <img
      alt=""
      className="styles__StudyGuidesCTAArrowDesktop-dthmzf-10 dfOdsD"
      src="arrow-desktop.svg"
    />
    <img
      alt=""
      className="styles__StudyGuidesCTAArrowMobile-dthmzf-11 jzZbVU"
      src="arrow-mobile.svg"
    />
  </div>
</div>
`;
